R/with.R

```diff
@@ -0,0 +1,125 @@
+#'                             assignment_linter = NULL,
+#'                             absolute_path_linter())
+#'
+#' # custom list of undesirable functions:
+#' #    remove sapply (using NULL)
+#' #    add cat (with a accompanying message),
+#' #    add print (unnamed, i.e. with no accompanying message)
+#' #    add return (as taken from all_undesirable_functions)
+#' my_undesirable_functions <- with_defaults(default = default_undesirable_functions,
+#'   sapply=NULL, "cat"="No cat allowed", "print", all_undesirable_functions[["return"]])
+#' @export
+with_defaults <- function(..., default = default_linters) {
+  vals <- list(...)
+  nms <- names2(vals)
+  missing <- !nzchar(nms, keepNA = TRUE)
+  if (any(missing)) {
+    args <- as.character(eval(substitute(alist(...)[missing])))
+    # foo_linter(x=1) => "foo"
+    # var[["foo"]]    => "foo"
+    nms[missing] <- re_substitutes(
+      re_substitutes(
+          # Very long input might have newlines which are not caught
+          #  by . in a perl regex; see #774
+        re_substitutes(args, rex("(", anything), "", options = "s"),
+        rex(start, anything, '["'),
+        ""
+      ),
+      rex('"]', anything, end),
+      ""
+    )
+  }
+
+  is.na(vals) <- nms == vals
+  default[nms] <- vals
+
+  res <- default[!vapply(default, is.null, logical(1L))]
+
+  res[] <- lapply(res, function(x) {
+    prev_class <- class(x)
+    if (inherits(x, "function") && !inherits(x, "lintr_function")) {
+      class(x) <- c(prev_class, "lintr_function")
+    }
+    x
+  })
+}
+
+#' Create a tag-based linter configuration
+#'
+#' Make a new list based on all linters tagged with `tags` from provided by `packages`.
+#' The result of this function is meant to be passed to the `linters` argument of `lint()`,
+#' or to be put in your configuration file.
+#'
+#' @param ... Arguments of elements to change. If unnamed, the argument is automatically named.
+#' If the named argument already exists in the list of linters, it is replaced by the new element.
+#' If it does not exist, it is added. If the value is `NULL`, the linter is removed.
+#' @inheritParams available_linters
+#'
+#' @return A modified list of linters.
+#' @seealso [with_defaults] for basing off lintrs set of default linters.
+#' @seealso [available_linters] to get a data frame of available linters.
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @examples
+#' # Default invocations of `with_defaults()` and `linters_with_tags()` are the same:
+#' all.equal(with_defaults(), linters_with_tags())
+#'
+#' # Get all linters useful for package development
+#' linters_with_tags(tags = "package_development")
+#'
+#' # Get all linters provided by lintr
+#' linters_with_tags(tags = NULL)
+#'
+#' # Get all linters tagged as "default" from lintr and mypkg
+#' \dontrun{linters_with_tags(packages = c("lintr", "mypkg"))}
+#' @export
+linters_with_tags <- function(..., tags = "default", packages = "lintr") {
```

Comment:
my sense is there should be no default for `tags=`. Maybe the signature should be switched, too, to allow `tags=` supplied positionally?

```
linters_with_tags <- function(tags, ..., packages = "lintr") 
```

Seeing

```
linters: linters_with_tags("default")
```

in config looks more natural to me.